pyproject.toml

```diff
@@ -6,7 +6,7 @@ authors = [
   { name="Faisal Dosani", email="faisal.dosani@capitalone.com" },
 ]
 license = {text = "Apache Software License"}
-dependencies = ["boto3<=1.35.9,>=1.9.92", "PyYAML<=6.0.1,>=5.1", "pandas<=2.2.2,>=0.25.2", "numpy<=2.0.2,>=1.22.0"]
+dependencies = ["boto3<=1.35.9,>=1.9.92", "PyYAML<=6.0.1,>=5.1", "pandas<=2.2.2,>=0.25.2", "numpy<=2.0.2,>=1.22.0", "polars>=1.5.0"]
```

Comment:
Is 1.5 actually the min version that works? I think it's worth looking into this as I think people are still using versions before 1.0.

Reply:
the minimum version needed is 1.0 for one of the methods used ([collect_schema](https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.collect_schema.html)) which is necessary for supporting both lazyframe and dataframe. It really draws down to if we want to support both hmm